ddtrace/internal/datadog/profiling/dd_wrapper/CMakeLists.txt

```diff
@@ -7,45 +7,64 @@ project(dd_wrapper
 # C++17
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Werror -Wextra -Wshadow")
-
-# If the compiler supports fanalyzer, let's also use that
-if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "10.0")
-        set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fanalyzer")
-    endif()
-endif()
 
 # Infer some basic properties about the build.  This is necessary because multiple
 # extensions reuse this cmake file, so we need its artifacts to go in a consistent
 # place
 get_filename_component(dd_wrapper_BUILD_PARENT "${CMAKE_BINARY_DIR}/../.." ABSOLUTE)
-set(dd_wrapper_BUILD_DIR "${dd_wrapper_BUILD_PARENT}/ddtrace.internal.datadog.profiling" PARENT_SCOPE)
+set(dd_wrapper_BUILD_DIR "${dd_wrapper_BUILD_PARENT}/ddtrace.internal.datadog.profiling")
+
+# Custom modules are in the parent directory
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../cmake")
 
 # Setup state for FindLibdatadog
 set(Datadog_BUILD_DIR "${dd_wrapper_BUILD_PARENT}/libdatadog")
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../cmake")
 
 # Includes
 include(CMakePackageConfigHelpers) # For generating the config file
 include(FetchContent)
 include(ExternalProject)
 include(FindLibdatadog)
-include(FindPython3)
 
 # Send some of these vars back up to the parent
-set(Datadog_INCLUDE_DIRS "${Datadog_INCLUDE_DIRS}" PARENT_SCOPE)
+if (DD_WRAPPER_IS_SUBPROJECT)
+  set(Datadog_INCLUDE_DIRS "${Datadog_INCLUDE_DIRS}" PARENT_SCOPE)
+  set(dd_wrapper_BUILD_DIR "${dd_wrapper_BUILD_PARENT}/ddtrace.internal.datadog.profiling" PARENT_SCOPE)
+endif()
+
+# Set some additional parameters
+add_compile_options(
+  "$<$<CONFIG:Debug>:-Og;-ggdb3;-fno-omit-frame-pointer>"
+  "$<$<CONFIG:Release>:-O3>"
+  -Wall -Werror -Wextra -Wshadow -Wnon-virtual-dtor -Wold-style-cast
+)
 
 # Library sources
 add_library(dd_wrapper SHARED
-    src/exporter.cpp
+    src/uploader_builder.cpp
+    src/sample_builder.cpp
+    src/global_cache.cpp
+    src/profile.cpp
+    src/uploader.cpp
+    src/sample.cpp
     src/interface.cpp
 )
 
+# If we were given flags to do sanitization or static analyis, use them
+if (SANITIZE_OPTIONS)
+  if (SANITIZE_OPTIONS MATCHES "fanalyzer")
+    target_compile_options(dd_wrapper PRIVATE -fanalyzer)
+  elseif (SANITIZE_OPTIONS MATCHES "cppcheck")
+    set(DO_CPPCHECK ON)
+  else()
```

Comment:
First of all, I have some uncertainty around the future of this precise code.  I believe what we'll end up doing is standardizing all of the logic in cmake and just calling specific arguments from CI rather than having a full-on wrapper to help build things.  That's what cmake is for.

As for your question, the legal values are

- the string "fanalyzer" which is only valid for GCC and enables GCC's `-fanalyze` static analysis flag
- the string "cppcheck" which will not add any sanitizers, but will build cppcheck (if needed) and run it afterward
- everything else is treated as a valid argument to the clang frontend's `-fsanitize=` option (and also as a linker flag); whether or not _those_ are valid depends on what [clang accepts](https://clang.llvm.org/docs/UsersManual.html#controlling-code-generation), which varies from build-to-build (for example, 17 has a leak analyzer which is separate from memsan).

Ultimately, I think we'll pin this to whatever features we guarantee in the build process.

Not sure if I answered the question, hit me back if I missed the point.